moveit_planners/ompl/ompl_interface/src/planning_context_manager.cpp

```diff
@@ -346,28 +346,8 @@ ompl_interface::ModelBasedPlanningContextPtr ompl_interface::PlanningContextMana
     // Choose the correct simple setup type to load
     context_spec.ompl_simple_setup_.reset(new ompl::geometric::SimpleSetup(context_spec.state_space_));
 
-    bool state_validity_cache = true;
-    if (config.config.find("subspaces") != config.config.end())
-    {
-      context_spec.config_.erase("subspaces");
-      // if the planner operates at subspace level the cache may be unsafe
-      state_validity_cache = false;
-      boost::char_separator<char> sep(" ");
-      boost::tokenizer<boost::char_separator<char> > tok(config.config.at("subspaces"), sep);
-      for (boost::tokenizer<boost::char_separator<char> >::iterator beg = tok.begin(); beg != tok.end(); ++beg)
-      {
-        const ompl_interface::ModelBasedStateSpaceFactoryPtr& sub_fact = factory_selector(*beg);
-        if (sub_fact)
-        {
-          ModelBasedStateSpaceSpecification sub_space_spec(robot_model_, *beg);
-          context_spec.subspaces_.push_back(sub_fact->getNewStateSpace(sub_space_spec));
-        }
-      }
-    }
-
     ROS_DEBUG_NAMED("planning_context_manager", "Creating new planning context");
     context.reset(new ModelBasedPlanningContext(config.name, context_spec));
-    context->useStateValidityCache(state_validity_cache);
```

Comment:
This line still seems relevant. Without it, `use_state_validity_cache_` in context might be uninitialized. Having said that, perhaps that member can be deleted as well, since it is always true with your changes.

Reply:
It is initialized in the constructor, but I agree that it can be removed.

This seems to imply that [ompl_interface::StateValidityChecker::isValidWithoutCache](https://github.com/ros-planning/moveit/blob/fe6498061103aaa299b0ef20d3e8a3dce39964d5/moveit_planners/ompl/ompl_interface/src/detail/state_validity_checker.cpp#L112) can also be removed (~70 lines of code). It is only used in [this](https://github.com/ros-planning/moveit/blob/fe6498061103aaa299b0ef20d3e8a3dce39964d5/moveit_planners/ompl/ompl_interface/src/detail/state_validity_checker.cpp#L74) context:
```C++
return planning_context_->useStateValidityCache() ? isValidWithCache(state, verbose) :
                                                      isValidWithoutCache(state, verbose);
```

I find it difficult to believe that all this code is not used by the current interface. Maybe I need to dig a bit in very commits to understand the reasoning behind this difference?

Reply:
> I find it difficult to believe that all this code is not used by the current interface. Maybe I need to dig a bit in very commits to understand the reasoning behind this difference?

As someone digging through this code base for years, I don't find it difficult to believe.
Even if there are still hooks for it around, they likely haven't been used by anyone in years.
Please go ahead and simplify the interface further if you can.